rclcpp/include/rclcpp/publisher_base.hpp

```diff
@@ -182,18 +181,12 @@ class PublisherBase
   using IntraProcessManagerSharedPtr =
     std::shared_ptr<rclcpp::intra_process_manager::IntraProcessManager>;
 
-  /// Implementation utility function that creates a typed mapped ring buffer.
-  RCLCPP_PUBLIC
-  mapped_ring_buffer::MappedRingBufferBase::SharedPtr
-  virtual make_mapped_ring_buffer(size_t size) const;
-
```

Comment:
Done

Reply:
As a note: this required substituting `dynamic_pointer_cast` with `static_pointer_cast` in `create_publisher.hpp`.

Reply:
If the destructor is not virtual then it will not call the destructor of the derived class when deleting a pointer of type `PublisherBase`. Also, if the destructor is not virtual and we don't need the destructor in `Publisher<T>` (not sure if we can get away from that) then we'd want to mark the class `final`, as it's no longer safe to casually inherit from it.

Probably, we should just leave the destructor `virtual`...